examples/06-fetch-map/legend.ts

```diff
@@ -0,0 +1,107 @@
+import {LayerDescriptor} from '@carto/api-client';
+import './legend.css';
+
+export function createLegend(layers: LayerDescriptor[]): HTMLElement {
+  const wrapper = document.createElement('div');
+  wrapper.className = 'legend-wrapper';
+
+  const container = document.createElement('div');
+  container.className = 'legend-container';
+
+  layers.forEach((layer) => {
+    const widgetSource = layer.props.data?.widgetSource?.props;
+    if (!widgetSource) return;
+
+    const {columns, spatialDataColumn} = widgetSource;
```

Comment:
I don't think this matters for examples in this repository, but one thing to note for the future fetchMap/Builder integration: I don't think we're consistently initializing data sources (e.g. `h3TilesetSource`) in Builder with the correct spatialDataColumn, it looks like sometimes that is inferred later, and we use `vectorTilesetSource` without knowing (yet) whether it's geometry or quadbin or h3. So we might need a different approach than this in Builder.